src/overlays/actors/ovl_En_Fish/z_en_fish.c

```diff
@@ -47,73 +63,709 @@ static ColliderJntSphInit D_80A17040 = {
+                     ANIMMODE_LOOP_INTERP, 2.0f);
+}
+
+void EnFish_BeginRespawn(EnFish* this) {
+    this->respawnTimer = 400;
+    Actor_SetScale(&this->actor, 0.001f);
+    this->actor.draw = NULL;
+}
+
+void EnFish_SetCutsceneData(EnFish* this) {
+    Actor* thisx = &this->actor;
+
+    if (D_80A17010 == NULL) {
+        D_80A17010 = thisx;
+        Actor_SetScale(thisx, 0.01f);
+        thisx->draw = EnFish_Draw;
+        thisx->shape.rot.x = 0;
+        thisx->shape.rot.y = -0x6410;
+        thisx->shape.rot.z = 0x4000;
+        thisx->shape.yOffset = 600.0f;
+        D_80A17014 = 10.0f;
+        D_80A17018 = 0.0f;
+        thisx->flags |= 0x10;
+        EnFish_SetOutOfWaterAnimation(this);
+    }
+}
+
+void EnFish_ClearCutsceneData(EnFish* this) {
+    D_80A17010 = NULL;
+    D_80A17014 = 0.0f;
+    D_80A17018 = 0.0f;
+}
+
+void EnFish_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnFish* this = THIS;
+    s16 params = this->actor.params;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gFishSkel, &gFishInWaterAnim, this->jointTable, this->morphTable,
+                       7);
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit, this->colliderItems);
+    this->actor.colChkInfo.mass = 50;
+    this->slowPhase = Rand_ZeroOne() * (0xFFFF + 0.5f);
+    this->fastPhase = Rand_ZeroOne() * (0xFFFF + 0.5f);
+
+    if (params == FISH_DROPPED) {
+        this->actor.flags |= 0x10;
+        ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 8.0f);
+        EnFish_Dropped_SetupFall(this);
+    } else if (params == FISH_SWIMMING_UNIQUE) {
+        EnFish_Unique_SetupSwimIdle(this);
+    } else {
+        EnFish_Respawning_SetupSlowDown(this);
+    }
+}
+
+void EnFish_Destroy(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnFish* this = THIS;
+
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+void EnFish_SetYOffset(EnFish* this) {
+    this->actor.shape.yOffset += ((Math_SinS(this->slowPhase) * 10.0f) + (Math_SinS(this->fastPhase) * 5.0f));
+    this->actor.shape.yOffset = CLAMP(this->actor.shape.yOffset, -200.0f, 200.0f);
+}
+
+s32 EnFish_InBottleRange(EnFish* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player = PLAYER;
+    Vec3f sp1C;
+
+    if (this->actor.xzDistToPlayer < 32.0f) {
+        sp1C.x = (Math_SinS(this->actor.yawTowardsPlayer + 0x8000) * 16.0f) + player->actor.world.pos.x;
+        sp1C.y = player->actor.world.pos.y;
+        sp1C.z = (Math_CosS(this->actor.yawTowardsPlayer + 0x8000) * 16.0f) + player->actor.world.pos.z;
+
+        //! @bug: this check is superfluous: it is automatically satisfied if the coarse check is satisfied. It may have
+        //! been intended to check the actor is in front of Player, but yawTowardsPlayer does not depend on Player's
+        //! world.rot.
```

Comment:
in this error message (and the other places it was copied to) I would just say "world rotation" instead of `world.rot`